src/InteractiveWindow/Editor/InteractiveWindow_UIThread.cs

```diff
@@ -214,12 +213,13 @@ public Task<ExecutionResult> ResetAsync(bool initialize)
                     _window._currentLanguageBuffer = null;
                 }
 
-                // replace the task being interrupted by a "reset" task:
                 State = State.Resetting;
-                _currentTask = _window._evaluator.ResetAsync(initialize);
-                _currentTask.ContinueWith(FinishExecute, _uiScheduler);
+                var executionResult = await _window._evaluator.ResetAsync(initialize).ConfigureAwait(true);
 
-                return _currentTask;
+                Debug.Assert(State == State.Resetting, $"Unexpected state {State}");
+                FinishExecute(executionResult.IsSuccessful);
```

Comment:
My understanding is that the exception will now be propagated, presumably triggering a (non-fatal?) Watson.  I guess I can induce a throw and see what happens.